layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,49 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+(defun spacemacs/move-buffer-to-window (windownum follow-focus-p)
+  "Moves a buffer to a window, using the spacemacs numbering. follow-focus-p
+   controls whether focus moves to new window (with buffer), or stays on
+   current"
+  (interactive)
+  (let ((b (current-buffer))
+        (w (selected-window)))
+    (save-selected-window
+      (select-window-by-number windownum)
+      (switch-to-buffer b)
+      (select-window w)
+      (switch-to-prev-buffer)
+      (unrecord-window-buffer (selected-window) b)))
+  (if follow-focus-p (select-window-by-number windownum)))
+
+(defun spacemacs/swap-buffers-to-window (windownum follow-focus-p)
+  "Swaps visible buffers between active window and selected window.
+   follow-focus-p controls whether focus moves to new window (with buffer), or
+   stays on current"
+  (interactive)
+  (let ((b1 (current-buffer))
+        (w1 (selected-window)))
+    (save-selected-window
```

Comment:
Also here `save-selected-window`+`select-window-by-number`+`switch-to-buffer` should be replaced by `get-window-by-number`+`set-window-buffer`.